android/src/main/java/com/swmansion/rnscreens/bottomsheet/DimmingFragment.kt

```diff
@@ -0,0 +1,482 @@
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        initViewHierarchy()
+        return containerView
+    }
+
+    override fun onViewCreated(
+        view: View,
+        savedInstanceState: Bundle?,
+    ) {
+        if (screen.sheetInitialDetentIndex <= screen.sheetLargestUndimmedDetentIndex) {
+            dimmingView.alpha = 0.0F
+        } else {
+            dimmingView.alpha = maxAlpha
+        }
+    }
+
+    override fun onStart() {
+        // This is the earliest we can access child fragment manager & present another fragment
+        super.onStart()
+        insetsProxy.registerOnView(requireRootView())
+        presentNestedFragment()
+    }
+
+    override fun onResume() {
+        insetsProxy.addOnApplyWindowInsetsListener(this)
+        super.onResume()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        insetsProxy.removeOnApplyWindowInsetsListener(this)
+    }
+
+    override fun onStop() {
+        super.onStop()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+    }
+
+    override fun onStateChanged(
+        source: LifecycleOwner,
+        event: Lifecycle.Event,
+    ) {
+        when (event) {
+            Lifecycle.Event.ON_START -> {
+                nestedFragment.screen.sheetBehavior?.let {
+                    dimmingViewCallback = AnimateDimmingViewCallback(nestedFragment.screen, dimmingView, maxAlpha)
+                    it.addBottomSheetCallback(dimmingViewCallback!!)
+                }
+            }
+            Lifecycle.Event.ON_STOP -> {
+                dismissSelf(emitDismissedEvent = true)
+            }
+            else -> {}
+        }
+    }
+
+    private fun presentNestedFragment() {
+        childFragmentManager.commit(allowStateLoss = true) {
+            setReorderingAllowed(true)
+            add(requireView().id, nestedFragment.fragment, null)
+        }
+    }
+
+    private fun removeNestedFragment() {
+        childFragmentManager.commit(allowStateLoss = true) {
+            setReorderingAllowed(true)
+            remove(nestedFragment.fragment)
+        }
+    }
+
+    private fun cleanRegisteredCallbacks() {
+        dimmingViewCallback?.let {
+            nestedFragment.screen.sheetBehavior?.removeBottomSheetCallback(it)
+        }
+        dimmingView.setOnClickListener(null)
+        nestedFragment.fragment.lifecycle.removeObserver(this)
+        insetsProxy.removeOnApplyWindowInsetsListener(this)
+    }
+
+    private fun dismissSelf(emitDismissedEvent: Boolean = false) {
+        if (!this.isRemoving) {
+            if (emitDismissedEvent) {
+                val reactContext = nestedFragment.screen.reactContext
+                val surfaceId = UIManagerHelper.getSurfaceId(reactContext)
+                UIManagerHelper
+                    .getEventDispatcherForReactTag(reactContext, screen.id)
+                    ?.dispatchEvent(ScreenDismissedEvent(surfaceId, screen.id))
+            }
```

Comment:
Definitely. I think we should group all event emitting code to some separate delegate object, so that the fragment / component files are much less cluttered. Won't do it here, this PR has already too many changes. Created a ticket for this.
https://github.com/orgs/software-mansion/projects/3/views/1?pane=issue&itemId=74218847